### PR TITLE
[#9356] Show loading spinner when fetching authentication details is ongoing

### PR DESCRIPTION
**Pull request by monmanuela**
Part of #9356 

**Outline of Solution**
How the page looks when the http request is incomplete:
image

I added a component for the loading spinner and use it in `admin-page`, `instructor-page`, `student-page` and `static-page`. Used a boolean flag to decide whether to show the loading spinner or not.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@monmanuela It may also be a good idea to add a loading indicator in the "Login" section at the upper right. It should only show user credentials/"Login" after it has been determined that there is/there is no logged in user.

**Comment by monmanuela** [comment]
@wkurniawan07 I get what you mean, but having a loading spinner on a nav bar looks quite weird. What about when it hasn't been determined yet, we just leave the top right-hand corner empty? It won't mislead the user and it won't look as weird

File: src/web/app/components/loading-spinner/loading-spinner.component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.loading-container {
+  margin: 0;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  -ms-transform: translate(-50%, -50%);
```
Comment (jacoblipech) [inline src/web/app/components/loading-spinner/loading-spinner.component.scss]: I think `-ms-` is applied to internet explorer right? Not necessary?

Comment (wkurniawan07) [reply src/web/app/components/loading-spinner/loading-spinner.component.scss]: > What about when it hasn't been determined yet, we just leave the top right-hand corner empty?

That works, too.

**Comment by wkurniawan07** [comment]
> What about when it hasn't been determined yet, we just leave the top right-hand corner empty?

That works, too.

File: src/web/app/components/loading-spinner/loading-spinner.component.scss (hunk 1 above)
Comment (wkurniawan07) [reply src/web/app/components/loading-spinner/loading-spinner.component.scss]: Actually, we're not likely to need any auto-prefixed version. Angular CLI comes with [Autoprefixer built-in](https://angular.io/guide/build#configuring-browser-compatibility).

Comment (monmanuela) [reply src/web/app/components/loading-spinner/loading-spinner.component.scss]: Got it!

**Comment by monmanuela** [comment]
> LGTM! Do you wan to try to use the `ajax-loading` component for loading the login part? I do agree that user shouldn't be able to see it when it is loading!

I tried to use the `ajax-loading` component, but I observed that the time taken to fetch data is less than time to load the gif haha. So I just leave it empty instead

**Pull request merged**